src/packages/@app/pages/Dashboard/Dashboard.ViewModel.ts

```diff
@@ -210,18 +239,47 @@ export class DashboardViewModel {
   };
 
   public refreshEnvironment = async (workspaceId) => {
-    const response =
-      await this.environmentService.fetchAllEnvironments(workspaceId);
-    if (response.isSuccessful && response.data.data) {
-      const environments = response.data.data;
-      await this.environmentRepository.refreshEnvironment(
-        environments,
-        workspaceId,
-      );
+    const isGuestUser = await this.getGuestUserState();
+    if (isGuestUser !== true) {
+      const response =
+        await this.environmentService.fetchAllEnvironments(workspaceId);
+      if (response.isSuccessful && response.data.data) {
+        const environments = response.data.data;
+        await this.environmentRepository.refreshEnvironment(
+          environments,
+          workspaceId,
+        );
+      }
     }
     return;
   };
 
+  /**
+   * clear local DB and clear guest user details from store
+   */
+  public clearLocalDB = async (): Promise<void> => {
+    setUser(null);
+    isGuestUserActive.set(false);
+    await this.tabRepository.clearTabs();
+    await this.guestUserRepository.clearTabs();
+    await RxDB.getInstance().destroyDb();
```

Comment:
`await this.guestUserRepository.clearTabs();`
??

Reply:
it was to clear the data of tabs although as we are destroying the db itself so it can be avoided. 

And lastly we are not providing the collection and environment for now that's why no restrictions has been added on api level.